### PR TITLE
Conditionally leverage Viewer web socket connection for Local Messaging

### PR DESCRIPTION
**Pull request by stulees**
- Determines websocket connection already open if `top.RiseVision.Viewer.LocalMessaging` exists in top level window. If exists:
  - Does not open a new connection and does not apply connection handlers to open connection
  - Leverages Viewers `write` and `receiveMessages` methods for messaging communication
  - `isConnected()` leverages Viewers `canConnect()` for status

### CODE REVIEW

**Comment by stulees** [comment]
@santiagonoguez I have validated this with a schedule running image demo and a presentation with Image & Video widgets using the staged version of this library. Working as expected.

File: src/rise-local-messaging.js [hunk 1]
```diff
@@ -46,23 +47,18 @@ RisePlayerConfiguration.LocalMessaging = (() => {
   }
 
   function _broadcastWebsocketMessage( msg ) {
+    if ( _useViewerWebsocketConnection ) {
+      top.RiseVision.Viewer.LocalMessaging.write( msg );
```
Comment (santiagonoguez) [inline src/rise-local-messaging.js]: @stulees in this case we are not assigning { from: _clientName }, is that something viewer does ?

File: src/rise-local-messaging.js [hunk 2]
```diff
@@ -89,24 +85,21 @@ RisePlayerConfiguration.LocalMessaging = (() => {
...
-    if ( !_isWebsocketConnectionAvailable()) {
-      console.log( "primus client side library was not loaded" );
-      return;
+  function _isWebsocketOpenByViewer() {
+    try {
+      if ( top.RiseVision && top.RiseVision.Viewer && top.RiseVision.Viewer.LocalMessaging ) {
+        return true;
+      }
+    } catch ( err ) {
+      console.log( "window.top reference error", err );
```
Comment (santiagonoguez) [inline src/rise-local-messaging.js]: I wonder if we should log this as an error to BQ, just to know that we are having CORS problems maybe. Thoughts ?

Comment (santiagonoguez) [reply src/rise-local-messaging.js]: sounds good then ! :D

**Pull request merged**